# Cleaning weekly M4 data

***

In [1]:
# import modules
import pandas as pd

In [2]:
# import training data for weekly time series
weekly_train = pd.read_csv("../Data/Train/Weekly-train.csv")

In [3]:
# import identifying file for time series
m4_info = pd.read_csv("../Data/M4-info.csv")

Using information from the M4 paper, we know there are 112 weekly level time series from the Micro domain. This seems like a decent number for initially implementing our models and methods.

In [4]:
# store ids for Weekly time series in Micro category
time_freq = "Weekly"
sector = "Finance"
ts_ids = m4_info.loc[(m4_info.SP == time_freq) & (m4_info.category == sector),:]["M4id"]

In [5]:
# subset weekly time series data for Micro time series
ts = weekly_train.loc[weekly_train.V1.isin(ts_ids),:]

In [6]:
# sum the number of NA values in each column
# use to remove any column with missing values
ts = ts.loc[:, ts.isna().sum() == 0]

In [7]:
ts.shape

(164, 248)

We are left with 164 series with 248 measurements in each. Save the file with the row ID's removed.

In [8]:
ts.iloc[:, 1:].to_csv("../Data/Train/Clean/weekly_finance_clean.csv", index = False)